astroquery/tests/test_cache.py

```diff
@@ -0,0 +1,222 @@
+import pytest
+import requests
+import os
+
+from time import sleep
+from pathlib import Path
+
+from astropy.config import paths
+
+from astroquery.query import QueryWithLogin
+from astroquery import conf
+
+URL1 = "http://fakeurl.edu"
+URL2 = "http://fakeurl.ac.uk"
+
+TEXT1 = "Penguin"
+TEXT2 = "Walrus"
+
+
+def set_response(resp_text, resp_status=200):
+    """Function that allows us to set a specific mock response for cache testing"""
+
+    def get_mockreturn(url, *args, **kwargs):
+        """Generate a mock return to a requests call"""
+
+        myresp = requests.Response()
+        myresp._content = resp_text
+        myresp.request = requests.PreparedRequest()
+        myresp.status_code = resp_status
+
+        return myresp
+
+    requests.Session.request = get_mockreturn
+
+
+class TestClass(QueryWithLogin):
+    """Bare bones class for testing caching"""
+
+    def test_func(self, requrl):
+
+        return self._request(method="GET", url=requrl)
+
+    def _login(self, username):
+
+        resp = self._request(method="GET", url=username)
+
+        if resp.content == "Penguin":
+            return True
+        else:
+            return False
+
+
+def test_conf():
+    conf.reset()
+
+    default_timeout = conf.cache_timeout
+    default_active = conf.cache_active
+
+    assert default_timeout == 604800
+    assert default_active is True
+
+    with conf.set_temp("cache_timeout", 5):
+        assert conf.cache_timeout == 5
+
+    with conf.set_temp("cache_active", False):
+        assert conf.cache_active is False
+
+    assert conf.cache_timeout == default_timeout
+    assert conf.cache_active == default_active
+
+    conf.cache_timeout = 5
+    conf.cache_active = False
+    conf.reset()
+
+    assert conf.cache_timeout == default_timeout
+    assert conf.cache_active == default_active
+
+
+def test_basic_caching():
+    conf.reset()
+
+    mytest = TestClass()
+    assert conf.cache_active
+
+    mytest.clear_cache()
+    assert len(os.listdir(mytest.cache_location)) == 0
+
+    set_response(TEXT1)
+
+    resp = mytest.test_func(URL1)
+    assert resp.content == TEXT1
+    assert len(os.listdir(mytest.cache_location)) == 1
```

Comment:
The following alternative avoids `os`, but is slightly longer:
```suggestion
    assert len(tuple(mytest.cache_location.iterdir())) == 1
```